File: src/webui/app/logic/domain/person-relationship.ts
import { PersonRelationship } from "jupiter-gen";

export function personRelationshipName(
  relationship: PersonRelationship
): string {
  switch (relationship) {
    case PersonRelationship.FAMILY:
      return "Family";
    case PersonRelationship.FRIEND:
      return "Friend";
    case PersonRelationship.ACQUAINTANCE:
      return "Acquaintance";
    case PersonRelationship.SCHOOL_BUDDY:
      return "School Buddy";
    case PersonRelationship.WORK_BUDDY:
      return "Work Buddy";
    case PersonRelationship.COLLEAGUE:
      return "Colleage";
    case PersonRelationship.OTHER:
      return "Other";
  }
}

File: src/core/jupiter/core/domain/concept/persons/person_relationship.py
"""The relationship the user has with a person."""
from functools import total_ordering

from jupiter.core.framework.value import EnumValue, enum_value


@enum_value
@total_ordering
class PersonRelationship(EnumValue):
    """The relationship the user has with a person."""

    FAMILY = "family"
    FRIEND = "friend"
    ACQUAINTANCE = "acquaintance"
    SCHOOL_BUDDY = "school-buddy"
    WORK_BUDDY = "work-buddy"
    COLLEAGUE = "colleague"
    OTHER = "other"

    def __lt__(self, other: object) -> bool:
        """Compare this with another."""
        if not isinstance(other, PersonRelationship):
            raise Exception(
                f"Cannot compare an entity id with {other.__class__.__name__}",
            )

        all_values = self.get_all_values()

        return all_values.index(self.value) < all_values.index(other.value)
